src/Instance.cc

```diff
@@ -222,3 +223,21 @@ Instance::WriteOurPid()
     debugs(50, Important(23), "Created " << TheFile);
 }
 
+/// Returns MD5 hash of the PID file name to make a uniq service name.
+/// It is called from Ipc::Mem::Segment::GenerateName(), Ipc::Port::MakeAddr()
+/// and Ipc::Port::CoordinatorAddr() functions.
+SBuf
+Instance::GetPidFilenameHash()
+{
+    cache_key hash[SQUID_MD5_DIGEST_LENGTH];
+    const auto name = PidFilenameCalc().c_str();
+
+    SquidMD5_CTX ctx;
+    SquidMD5Init(&ctx);
+    SquidMD5Update(&ctx, (unsigned char*) name, strlen(name));
```

Comment:
The hash function does not need to care about c-string termination. Direct use of the `name` buffer is safe here.

```suggestion

    SquidMD5_CTX ctx;
    SquidMD5Init(&ctx);
    const auto name = PidFilenameCalc();
    SquidMD5Update(&ctx, name.rawContent(), name.length());
```